Sources/OpenAPIServiceModel/ParseSchemas.swift

```diff
@@ -0,0 +1,202 @@
+                                                                                                        maximum: integerContext.maximum?.value,
+                                                                                                        exclusiveMinimum: integerContext.minimum?.exclusive ?? false,
+                                                                                                        exclusiveMaximum: integerContext.maximum?.exclusive ?? false))
+            }
+            
+        case .object(_ , let objectContext):
+            if case .b(let mapSchema) = objectContext.additionalProperties {
+                parseMapDefinitionSchema(mapSchema: mapSchema,
+                                         enclosingEntityName: &enclosingEntityName,
+                                         model: &model)
+            } else {
+                var structureDescription = StructureDescription()
+                parseObjectSchema(structureDescription: &structureDescription, enclosingEntityName: &enclosingEntityName,
+                                  model: &model, objectContext: objectContext, modelOverride: modelOverride, document: document)
+                
+                model.structureDescriptions[enclosingEntityName] = structureDescription
+            }
+        case .array(_, let arrayContext):
+            parseArrayDefinitionSchemas(arrayMetadata: arrayContext, enclosingEntityName: &enclosingEntityName,
+                                        model: &model, modelOverride: modelOverride, document: document)
+        case .string(_, let stringContext):
+            addStringField(metadata: stringContext, schema: schema,
+                           model: &model, fieldName: enclosingEntityName, modelOverride: modelOverride)
+        case .number(_, let numberContext):
+            model.fieldDescriptions[enclosingEntityName] = Fields.double(rangeConstraint:
+                                                                            NumericRangeConstraint<Double>(minimum: numberContext.minimum?.value,
+                                                                                                           maximum: numberContext.maximum?.value,
+                                                                                                           exclusiveMinimum: numberContext.minimum?.exclusive ?? false,
+                                                                                                           exclusiveMaximum: numberContext.maximum?.exclusive ?? false))
+        case .all(let otherSchema, _), .any(let otherSchema, _), .one(let otherSchema, _):
+            var structureDescription = StructureDescription()
+            parseOtherSchemas(structureDescription: &structureDescription, enclosingEntityName: &enclosingEntityName, model: &model, otherSchema: otherSchema, modelOverride: modelOverride, document: document)
```

Comment:
nit: this line is a bit too long, can you put some of the variables on newlines?

Reply:
Done.